h/schemas/forms/admin/group.py

```diff
@@ -24,56 +22,65 @@
 
-        The creator of a group must belong to the same authority as the group
-        and the group's organization.  Validate that there is a user matching
-        the given creator username with the same authority as the chosen
-        organization.
+    The creator and members of a group must belong to the same authority as the
+    group and the group's organization.
+    """
+    user_svc = form.bindings["user_svc"]
 
-        """
-        user_svc = kw["user_svc"]
+    # Get the authority from the list of organisations based on the one
+    # actually picked in the form
+    authority = form.bindings["organizations"][value["organization"]].authority
 
-        # A {pubid: models.Organization} dict of all the organizations
-        # available to choose from in the form.
-        organizations = kw["organizations"]
+    exc = colander.Invalid(form, None)
 
-        # The pubid of the organization that the user has selected in the form.
-        selected_pubid = value["organization"]
+    _validate_members(form, value["members"], authority, user_svc, exc)
+    _validate_creator(value["creator"], authority, user_svc, exc)
 
-        # The models.Organization object for the selected organization.
-        selected_organization = organizations[selected_pubid]
+    if exc.children:
+        raise exc
 
-        # The authority that the new group will belong to if it is created.
-        authority = selected_organization.authority
 
-        # The username string that was entered for the group creator.
-        creator_username = value["creator"]
+def _validate_members(form, members_value, authority, user_svc, parent_exc):
+    if not members_value:
+        return
 
-        # The models.User object for the group creator user, or None.
-        user = user_svc.fetch(creator_username, authority)
+    members_node = form.get("members")
+    member_node = members_node.get("member")
+    members_error = colander.Invalid(members_node, None)
 
-        if not user:
-            # Either the username doesn't exist at all, or it has a different
-            # authority than the chosen organization.
-            exc = colander.Invalid(form, _("User not found"))
-            exc["creator"] = _(
-                "User {creator} not found at authority {authority}"
-            ).format(creator=creator_username, authority=authority)
-            raise exc
+    for pos, node_value in enumerate(members_value):
+        if not user_svc.fetch(node_value, authority):
+            # If we don't attach errors in the right position, they don't
+            # appear. For a list type value the correct position is the index
+            # of the value in the list
+            members_error.add(
+                colander.Invalid(
+                    member_node,
+                    _("User '{username}' not found at authority '{authority}'").format(
+                        username=node_value, authority=authority
+                    ),
```

Comment:
We now use the same error message for members as the creator, which includes the authority.